engine/src/main/java/io/zeebe/engine/state/instance/DbElementInstanceState.java

```diff
@@ -52,6 +56,19 @@
   private final ColumnFamily<DbLong, AwaitProcessInstanceResultMetadata>
       awaitProcessInstanceResultMetadataColumnFamily;
 
+  private final DbLong flowScopeKey = new DbLong();
+  private final DbString gatewayElementId = new DbString();
+  private final DbString sequenceFlowElementId = new DbString();
+  private final DbInt numberOfTakenSequenceFlows = new DbInt();
+  private final DbCompositeKey<DbLong, DbString> flowScopeKeyAndElementId;
+  private final DbCompositeKey<DbCompositeKey<DbLong, DbString>, DbString>
+      numberOfTakenSequenceFlowsKey;
+  /**
+   * [flow scope key | gateway element id | sequence flow id] => [number of taken sequence flows]
```

Comment:
I find this "[number of taken sequence flows]" a little confusing because the getter is also called `getNumberOfTakenSequenceFlows`. The way I understand it, an entry with that key counts how many times this particular sequence flow was taken, whereas the getter counts how many entries exist.

```suggestion
   * [flow scope key | gateway element id | sequence flow id] => [times the sequence flow was taken]
```

Reply:
Same with other methods/variables in this class